Dev/henry/10036 memberships update (#920)
datatrails/datatrails-docs: content/developers/api-reference/members-api/index.md

If you are looking for a simple way to test our API you might prefer our [Postman collection](https://www.postman.com/datatrails-inc/workspace/datatrails-public/overview), the [YAML runner](/developers/yaml-reference/story-runner-components/) or the [Developers](https://app.datatrails.ai) section of the web UI. 

Additional YAML examples can be found in the articles in the [Overview](/platform/overview/introduction/) section.
{{< /note >}}

## Members API Examples

Create the [bearer_token](/developers/developer-patterns/getting-access-tokens-using-app-registrations) and store in a file in a secure local directory with 0600 permissions.

### Listing members of your tenant

To fetch the list of members, simply `GET` the `/members` resource

```bash
curl -v -X GET \
     -H "@$HOME/.datatrails/bearer-token.txt" \
     https://app.datatrails.ai/archivist/iam/v1/members
```

By default this doesn't show members that are deactivated. You can filter the list to include 
deactivated users:

```bash
curl -v -X GET \
     -H "@$HOME/.datatrails/bearer-token.txt" \
     https://app.datatrails.ai/archivist/iam/v1/members?member_state=MEMBER_STATE_BOTH
```

> ### Promoting a member of your tenant

Reviewer: ```suggestion
### Promoting a user to 'OWNER' role in a tenancy
```

> You can manage the roles of a user (i.e. promoting or demoting them by way of the 'OWNER' role) by

Reviewer 2: ```suggestion
You can manage the roles of a member (i.e. promoting or demoting them by way of the 'OWNER' role) by
```

Reviewer 2: its a bit jarring how the UI refers to these as users, but the API as members? maybe we want to be consistent at least on a customer facing level? if so we can disregard this suggestion?

Author: What are your thoughts on this one @JAG-UK ? I know using the term 'Users' in the UI was requested. Is it worth the inconsistency overall?

Reviewer: They're conceptually different: a user having roles is about what a person (or piece of software) can do within the system. The way we express and manage that is through Tenancy Membership. So a 'user' is a 'member' of a number of 'tenancies'.

I'll make a suggestion in a separate comment

Author: Users only have roles in the context of a particular tenant though, i.e. their membership

> You can manage the roles of a user (i.e. promoting or demoting them by way of the 'OWNER' role) by
> PATCHing their record. With the identity of a member in your tenant to hand, submit this request
> to set their roles to a list containing only 'OWNER'. To demote the member, simply send an empty
> list.

Reviewer: ```suggestion
You can manage the roles of a user (i.e. promoting or demoting them with the 'OWNER' role) by
PATCHing their membership record. 
With the identity of the tenant member record corresponding to the user in question, submit this request
to set their roles to a list containing only 'OWNER'. To demote the member, simply send an empty
list.
```

```bash
curl -v -X PATCH \
    -H "@$HOME/.datatrails/bearer-token.txt" \
    -H "Content-type: application/json" \
    -d "{'roles': ['OWNER']}" \
    https://app.datatrails.ai/archivist/iam/v1/{member_identity}
```

### Deactivating a member of your tenant

> When a user should no longer have access to your tenant, you can deactivate them progamatically 

Reviewer 2: ```suggestion
When a member should no longer have access to your tenant, you can deactivate them progamatically 
```

> When a user should no longer have access to your tenant, you can deactivate them progamatically 

Reviewer 2: ```suggestion
When a user should no longer have access to your tenant, you can deactivate them programatically 
```
